### PR TITLE
Add grant command

### PR DESCRIPTION
**Pull request by emile-00**
Added grant command.

## Info
- **Description**: 
Document GRANT command.

- **Related code PR**: 
[ https://github.com/risingwavelabs/risingwave/pull/3004 ]

- **Related doc issue**: 
Resolves [ https://github.com/risingwavelabs/risingwave-docs/issues/422]

- **Notes**: 
[ Any additional information? ]

## Before merging
  - [ ] (For version-specific PR) I have selected the corresponding software version in **Milestone** and linked the related doc issue to this PR in **Development**.
  - [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`). 
  - [ ] I have checked the doc site preview, and the updated parts look good. <details><summary>How?</summary>Scroll down and open this link: image
</details>

## Published doc pages
  [ After merging this PR, edit the description to include the links to the updated doc pages here. For example, https://www.risingwave.dev/docs/latest/intro/. ]

### CODE REVIEW

File: docs/sql/commands/sql-grant.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+---
+
+Use the `GRANT` command to give a user specific privileges.
+
+## Syntax
+
+Grant a user database privileges.
+```sql
+GRANT {{CONNECT | CREATE}[, ...]| ALL [PRIVILEGES]} 
+ON DATABASE <db_name>
```
Comment (yezizp2012) [inline docs/sql/commands/sql-grant.md]:
```suggestion
ON DATABASE <db_name> [, ...]
```

Granting multiple object privileges to users in the same sql is supported as well.

File: docs/sql/commands/sql-grant.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+```sql
+GRANT {{CONNECT | CREATE}[, ...]| ALL [PRIVILEGES]} 
+ON DATABASE <db_name> 
+TO <user_name> [WITH GRANT OPTION] [GRANTED BY <user_name>];
+```
+
+Grant a user materialized view privileges.
+```sql
+GRANT {SELECT | ALL [PRIVILEGES]} 
+ON MATERIALIZED VIEW <mv_name>
```
Comment (yezizp2012) [inline docs/sql/commands/sql-grant.md]:
```suggestion
ON {MATERIALIZED VIEW <mv_name> [, ...] 
    | ALL MATERIALIZED VIEWS IN SCHEMA <schema_name> [, ...] }
```
Granting all materialized views in schema is also supported.

File: docs/sql/commands/sql-grant.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+```sql
+GRANT {SELECT | ALL [PRIVILEGES]} 
+ON MATERIALIZED VIEW <mv_name> 
+TO <user_name> [WITH GRANT OPTION] [GRANTED BY <user_name>];
+```
+
+Grant a user schema privileges.
+```sql
+GRANT {CREATE | ALL [PRIVILEGES]} 
+ON SCHEMA <schema_name>
```
Comment (yezizp2012) [inline docs/sql/commands/sql-grant.md]: Ditto.

File: docs/sql/commands/sql-grant.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+```sql
+GRANT {CREATE | ALL [PRIVILEGES]} 
+ON SCHEMA <schema_name> 
+TO <user_name> [WITH GRANT OPTION] [GRANTED BY <user_name>];
+```
+
+Grant a user source privileges.
+```sql
+GRANT {{SELECT | UPDATE | INSERT | DELETE} [, ...]| ALL [PRIVILEGES]} 
+ON SOURCE <source_name>
```
Comment (yezizp2012) [inline docs/sql/commands/sql-grant.md]: Ditto, same as mview.

**Review by yezizp2012 — COMMENTED** [review]
Rest LGTM

File: docs/sql/commands/sql-grant.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+```sql
+GRANT {CREATE | ALL [PRIVILEGES]} 
+ON SCHEMA <schema_name> [, ...]
+TO <user_name> [WITH GRANT OPTION] [GRANTED BY <user_name>];
+```
+
+Grant a user source privileges.
+```sql
+GRANT {{SELECT | UPDATE | INSERT | DELETE} [, ...]| ALL [PRIVILEGES]} 
+ON SOURCE <source_name> [, ...]
```
Comment (yezizp2012) [inline docs/sql/commands/sql-grant.md]:
```suggestion
ON {SOURCE <source_name> [, ...]
    | ALL SOURCES IN SCHEMA <schema_name> [, ...] }
```

**Review by yezizp2012 — COMMENTED** [review]
Rest LGTM.

File: docs/sql/commands/sql-grant.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+---
+
+Use the `GRANT` command to give a user specific privileges.
+
+## Syntax
+
+Grant a user database privileges.
+```sql
+GRANT {{CONNECT | CREATE}[, ...]| ALL [PRIVILEGES]} 
+ON DATABASE <db_name> [, ...]
```
Comment (hengm3467) [inline docs/sql/commands/sql-grant.md]: @emile-00 As a general practice, we just spell out variables like db_name and do not use angle brackets (<>) to enclose them. They'll be processed as in italic in the output. Please make the same changes (removing angle brackets around variables) across the whole topic. Thanks.

**Review by hengm3467 — APPROVED** [review]

**Pull request merged**